File: game-frontend/src/main/ts/networked/BulletInfo.ts
/**
 * Networked class carrying bullet info from the server to the client
 */
class BulletInfo {
    public owner: string;
    public state: number;
    public bulletId: number;
    public velocityX: number;
    public velocityY: number;
    public positionX: number;
    public positionY: number;

    static getValidArrayFromJson(json: string): BulletInfo[] {
        let length: number = json.length;

        //"BulletInfo[]:" is 13 characters long
        let beginning: string = json.substr(0, Math.min(length, 13));
        if(beginning === "BulletInfo[]:") {
            let jsonStr: string = json.substring(13, length);
            return JSON.parse(jsonStr) as BulletInfo[];
        }
        else {
            return null;
        }
    }
}

File: game-backend/src/main/java/com/becky/networked/message/BulletInfo.java
package com.becky.networked.message;

/**
 * The class of data to be sent to clients
 */
public class BulletInfo {
    private final String owner;
    private final int state;
    private final long bulletId;
    private final Float velocityX;
    private final Float velocityY;
    private final Float positionX;
    private final Float positionY;

    public BulletInfo(final String owner, final int state, final long bulletId,
                      final Float velocityX, final Float velocityY,
                      final Float positionX, final Float positionY) {
        this.owner = owner;
        this.state = state;
        this.bulletId = bulletId;
        this.velocityX = velocityX;
        this.velocityY = velocityY;
        this.positionX = positionX;
        this.positionY = positionY;
    }

    public String getOwner() {
        return this.owner;
    }

    public int getState() {
        return this.state;
    }

    public Float getVelocityX() {
        return this.velocityX;
    }

    public Float getVelocityY() {
        return this.velocityY;
    }

    public Float getPositionX() {
        return this.positionX;
    }

    public Float getPositionY() {
        return this.positionY;
    }

    public long getBulletId() {
        return this.bulletId;
    }
}
